attest/verifier/config/src/lib.rs

```diff
@@ -0,0 +1,449 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! This crate defines a serialization format for collections of enclave
+//! measurements and associated hardening advisories and other data, which can
+//! be used to configure an attestation verifier appropriately.
+
+#![no_std]
+#![deny(missing_docs)]
+#![allow(non_snake_case)]
+
+extern crate alloc;
+
+use alloc::{borrow::ToOwned, collections::BTreeMap, string::String, vec::Vec};
+use displaydoc::Display;
+use mc_attest_core::{MrEnclave, MrSigner};
+use mc_attest_verifier::{MrEnclaveVerifier, MrSignerVerifier, Verifier, DEBUG_ENCLAVE};
+use serde::{Deserialize, Serialize};
+use serde_hex::{SerHex, Strict};
+
+/// Defines a json schema for an individual attestation status verifier.
+/// This is either a MRENCLAVE or MRSIGNER type, and the type is inferred by
+/// the presence of these fields.
+/// Unknown fields are flagged as an error.
```

Comment:
the reasoning was to try to prevent a typo from silently allowing a bad config to be accepted.

for instance, if you had

```json
{
   "MRENCLAVE": "fffffff",
   "allow_hardening_advisorys": ["INTEL-00607", "INTEL-00666"]
}
```

with `deny_unknown_fields`, this will alert you to the typo `allow_hardening_advisorys != allow_hardening_advisories`, while without that, I think it would silently ignore the listed advisories and default the list to the empty list.

Reply:
I can see that. I can also see that if a new field gets added it might be strongly suggested for security reasons.

On the counter it feels like it's locking down the interface.
i.e. someone who has an old tool version using a newer config file may need to play whackamole removing unknown fields

i'm fine either way, only time will tell based on how customers utilize this interface

Reply:
Yeah -- what i think is likely to actually happen is, this file is shipped with each version of the client, so there likely isn't a reason to update the file without also updating the client.

(I suspect that SDK's are going to bake the json blob in as a string constant)

So I feel like it's okay to deny unknown fields, I feel like that's more likely to prevent a problem than to create one. But yeah, it's hard to be completely sure